### ISSUE TITLE
How to run default loaders from custom plugin?

### ISSUE THREAD

**Issue by JWo1F** [evanw/esbuild#3244]
Hi, First of all I want to thank you for such a fast and customizable library. It allows me to do a lot of things.

## Problem
At the moment I am working on generating static websites, the source code of which is written in typescript and react.
As an input point I need to set some `.tsx` file, which should turn into an `.html` file on output. I can do this by redirecting the path in the `onResolve` hook. However, I need to import the styles file and the original `.ts` file. It seems that I can't just read the `.ts` file with `fs`, because it has to be processed with the standard ts loader. But if I return that ts file in `onLoad`, that's what goes into the final output.

At the moment I have to subscribe to the onEnd hook and do my manipulations in it, which gives me 2 files: `.ts` and `.html`, but I would like to have only one.

## Suggestion
What I think can be done: by analogy with `build.resolve` make `build.load`, which will call the hook `onResolve`, then `onLoad` and then return the final result.
I'm also considering calling `build.esbuild.transorm()`, but in that case I get problems with assets.
Calling `build.esbuild.build()` would also cause issues with assemblies and break the links between builds, since it would actually be calling N+1 isolated builds.

**Comment by hyrious** [comment]
I can see your point of **reusing** a bundle result to build another file. However talking about SSR, this approach has some issues:

- The component code in SSR is supposed to run at server side, however in current `build` it is built for browser. Some simple cases may run, but when you want to add some options like different env variables it has to split.
- Sometimes you still have to make a real file to run a script. I mean `import(toDataUrl(code))` may not always work. For example when there's code reading `import.meta.url`. In the end you still have to write it to somewhere like `node_modules/.ssr/build.js`.

It happens that I did make [a site](https://hyrious.me/esbuild-repl/) with SSR enabled using esbuild. I'd like to share the approach: https://github.com/hyrious/esbuild-repl/blob/main/scripts/build.ts

1. Put the template html in entrypoints `entryPoints: ['src/index.html', 'src/index.ts']`. Later I will use a plugin and the `copy` loader to generate an SSR-ed file.
2. In the [SSR plugin](https://github.com/hyrious/esbuild-repl/blob/main/scripts/plugins/svelte-ssr.ts), I can read the template file path and the entry component file through plugin options or by analyzing the template.
3. Build (compile) the entry component as an SSR module and save it somewhere, then import it to generate the html fragment.
4. Inject the html into template, and return the final HTML with the `copy` loader.

**Comment by JWo1F** [comment]
@hyrious, yeah, I'm using something like your example. But unlike you, I don't need to run my code on the frontend. So I deliver bare html + vanilla javascript to the frontend. In this case it would be useful to be able to run load separately.

Thanks for the example, I will make some changes to my code inspired by your code!
